modules/images/webp-uploads/generation.php

```diff
@@ -0,0 +1,236 @@
+
+	// Bail if the current request is not to a subsize.
+	if ( ! preg_match( '#/(.+)-(\d+)x(\d+)\.([a-zA-Z]+)$#', $relpath, $matches ) ) {
+		return;
+	}
+
+	$img_width  = intval( $matches[2] );
+	$img_height = intval( $matches[3] );
+
+	$size       = '';
+	$size_names = get_intermediate_image_sizes();
+	$sizes      = wp_get_additional_image_sizes();
+
+	// Find the requested subsize name.
+	foreach ( $size_names as $size_name ) {
+		$width = intval(
+			isset( $sizes[ $size_name ] )
+				? $sizes[ $size_name ]['width']
+				: get_option( "{$size_name}_size_w" )
+		);
+
+		$height = intval(
+			isset( $sizes[ $size_name ] )
+				? $sizes[ $size_name ]['height']
+				: get_option( "{$size_name}_size_h" )
+		);
+
+		// Try to normalize dimensions if only one part matches for the current size.
+		if ( $width > 0 && $img_width !== $width && $img_height === $height ) {
+			$dimensions = wp_constrain_dimensions( $width, $height, 0, $img_height );
+			$img_width  = $dimensions[0];
+		} elseif ( $img_width === $width && $img_height !== $height && $height > 0 ) {
+			$dimensions = wp_constrain_dimensions( $width, $height, $img_width, 0 );
+			$img_height = $dimensions[1];
+		}
+
+		if (
+			// If the current size exactly matches requested width and height.
+			( $width === $img_width && $height === $img_height ) ||
+			// If the current size matches the requested height and has a variable width.
+			( 0 === $width && $height === $img_height ) ||
+			// If the current size matches the requested width and has a variable height.
+			( $width === $img_width && 0 === $height )
+		) {
+			$size = $size_name;
+
+			// Stop finding the size name if the current size exactly matches requested width and height.
+			// If the current size has a variable width or height, we need to continue finding in case there
+			// is an exact match.
+			if ( $width && $height ) {
+				break;
+			}
+		}
+	}
+
+	// Bail if the subsize name is not found.
+	if ( empty( $size ) ) {
+		return;
+	}
+
+	// Bail if we can't find an attachment with requested file.
+	$attachment_id = attachment_url_to_postid( "{$uploads_dir['baseurl']}/{$matches[1]}.{$matches[4]}" );
+	if ( empty( $attachment_id ) ) {
+		return;
+	}
+
+	// Bail if we can't find attachment metadata.
+	$metadata = wp_get_attachment_metadata( $attachment_id );
+	if ( empty( $metadata ) ) {
+		return;
+	}
+
+	require_once ABSPATH . '/wp-admin/includes/image.php';
+
+	// Generate missing sub size.
+	$metadata = _wp_make_subsizes(
+		array( $size => $sizes[ $size ] ),
+		"{$uploads_dir['basedir']}/{$matches[1]}.{$matches[4]}",
+		$metadata,
+		$attachment_id
+	);
+
+	// Generate sources.
+	$metadata = webp_uploads_create_sources_property( $metadata, $attachment_id );
+
+	// Save updated metadata.
+	wp_update_attachment_metadata( $attachment_id, $metadata );
+
+	$image_file = $uploads_dir['basedir'] . $relpath;
+
+	// Send headers if headers haven't been sent yet.
+	if ( ! headers_sent() ) {
```

Comment:
How about flipping and adding an early return instead?

```suggestion
	if ( headers_sent() ) {
	  return
	}
```

Reply:
Yep, works for me. Updated.